src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -190,22 +205,39 @@ private static void DumpProjectRestoreInfo(PackageSpec packageSpec)
                 var dgPath = Path.Combine(outputPath, $"{Guid.NewGuid()}.dg");
                 dgFile.Save(dgPath);
             }
-            catch(Exception ex)
+            catch (Exception e)
             {
-                ExceptionHelper.WriteToActivityLog(ex);
+                _logger.LogError(e.ToString());
             }
         }
 
         private static PackageSpec ToPackageSpec(ProjectNames projectNames, IVsProjectRestoreInfo projectRestoreInfo)
         {
-            var tfis = projectRestoreInfo.TargetFrameworks
+            var tfis = projectRestoreInfo
+                .TargetFrameworks
                 .Cast<IVsTargetFrameworkInfo>()
                 .Select(ToTargetFrameworkInformation)
                 .ToArray();
 
             var projectFullPath = Path.GetFullPath(projectNames.FullName);
             var projectDirectory = Path.GetDirectoryName(projectFullPath);
 
+            var originalTargetFrameworks = tfis
+                .Select(tfi => tfi.FrameworkName.GetShortFolderName())
+                .ToList();
+            var crossTargeting = originalTargetFrameworks.Count > 1;
```

Comment:
OriginalTargetFrameworks needs to use the exact strings from the ``$(TargetFrameworks)`` property, not strings formatted by nuget.

CrossTargeting should be true if ``$(TargetFrameworks)`` is not empty. Having a single framework here still counts as cross targeting.

If counting the target frameworks could determine if cross targeting was needed this property wouldn't exist :smile: